middle_end/closure/closure.ml

```diff
@@ -63,7 +63,16 @@ let rec build_closure_env env_param pos = function
    contain the right names if the -for-pack option is active. *)
 
 let getglobal dbg id =
-  Uprim(P.Pread_symbol (Compilenv.symbol_for_global id), [], dbg)
+  assert (Ident.is_global_or_predef id);
+  let pack_prefix =
+    if Ident.is_global id then Compilenv.pack_prefix_for_global_ident id
+    else Compilation_unit.Prefix.empty
+  in
+  let symbol =
+    Symbol.for_ident id ~pack_prefix
+    |> Symbol.linkage_name
+  in
```

Comment:
This code really doesn't belong here - almost exactly the same code needs to be added to `Opttoploop` for the same purpose: translating a global `Ident.t` to a linkage name. `Compilenv.symbol_for_global` used to do this, and `Compilenv` still needs to be consulted anyway, so we should just keep `Compilenv.symbol_for_global` after all (with this new implementation).

Reply:
I'm not sure about this (one thing is that Flambda 1 doesn't seem to need such an interface in `Compilenv`).  When `Ident.t` knows the `Compilation_unit.t`, `Compilenv` won't need to be involved, so it may be better to tolerate the code duplication in this intermediate state of the work.

Reply:
I'd still rather it be in just one place, even if we're about to get rid of it. It doesn't need to be in `Compilenv` necessarily.